### PR TITLE
windows: enable UnixSockets with all build toolchains

### PR DESCRIPTION
**Pull request by vszakats**
Extend existing unix socket support in Windows builds to be enabled regardless of the build toolchain vendor or version.

Ref: https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/
Ref: https://github.com/curl/curl/issues/5162

### CODE REVIEW

File: lib/curl_addrinfo.c [hunk 1]
```diff
@@ -50,7 +50,7 @@
 #  define in_addr_t unsigned long
 #endif
 
-#if defined(WIN32) && defined(USE_UNIX_SOCKETS)
+#if defined(USE_UNIX_SOCKETS) && !defined(WIN32)
 #include <afunix.h>
```
Comment (bagder) [inline lib/curl_addrinfo.c]: This header is not generically available (on non-Windows) so this causes build failures all over. I suspect this is actually a header that's introduced in some recent Windows sdk that enables unix domain sockets?

Comment (vszakats) [reply lib/curl_addrinfo.c]: Oh, indeed, I’ll submit an update in a few.

**Comment by bagder** [comment]
does @MarcelRaad or @mback2k have any opinion or thumbs up on this?

**Comment by mback2k** [comment]
The only problem I can see is that even if it is compiled into curl, the socket functionality will just straight out fail on older Windows systems. I hope that it does not produce a crash on those old systems.

**Comment by bagder** [comment]
That's not new since we already built it enabled in some builds but yeah, it would be good if someone on an older Windows could try it out and see what happens...

**Comment by vszakats** [comment]
In case we'd decide needing extra logic to fail more gracefully at runtime, it'd be best to implement it in a separate commit. I tend to think a short note in the documentation would work even better – there are already platform-specific notes there.

**Comment by bagder** [comment]
> it'd be best to implement it in a separate commit

Agreed. It is certainly separate from this PR.

**Comment by vszakats** [comment]
According to the calendar we're in feature freeze, but since there is no new feature here only the build-time enabling of an existing one, I was wondering if I shall commit this now or after the upcoming release?

**Comment by bagder** [comment]
I consider this a plain bugfix and think you can land this now!

**Comment by bagder** [comment]
Does this mean that #4040 is now also fixed - ie this is also enabled in cmake builds? (That issue listed in `KNOWN_BUGS`)

**Comment by vszakats** [comment]
@bagder Yes, by the looks of it, it should be fixed.

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**